internal/install/recipes/recipe_detector.go

```diff
@@ -43,16 +42,17 @@ func NewRecipeDetector(contex context.Context, repo *RecipeRepository) *RecipeDe
 	}
 }
 
-func (dt *RecipeDetector) detectRecipes() error {
-	if dt.isDetectionRunned {
-		return nil
-	}
+func (dt *RecipeDetector) GetDetectedRecipes() (map[string]*RecipeDetectionResult,
+	map[string]*RecipeDetectionResult,
+	error) {
 
-	dt.isDetectionRunned = true
+	if len(dt.availableRecipes) != 0 || len(dt.unavaliableRecipes) != 0 {
+		return dt.availableRecipes, dt.unavaliableRecipes, nil
+	}
```

Comment:
How could these two maps already have recipes?  `GetDetectedRecipes` only seems to be called from one place and appears responsible for setting these maps 🤔

Reply:
Updated